### PR TITLE
Event Forms & Filters

### PR DESCRIPTION
**Pull request by LatidoReMe**
+ Added filter form with type, unit, and version filtering
+ Fixed version cuteform
+ Fixed forms labels making extra translated terms
+ Adjusted VERSIONS dictionary to use images instead of icons
+ Made account creation show version images instead of icons
+ Other minor improvements/changes

### CODE REVIEW

File: sukutomo/forms.py [hunk 1]
```diff
@@ -47,24 +48,42 @@ class Meta:
         fields = ('search', 'i_attribute', 'i_unit', 'i_subunit', 'i_school', 'i_year', 'i_astrological_sign', 'i_blood')
 
 class EventForm(AutoForm):
-    start_date = forms.forms.DateField(label=_('Beginning'), required=False)
-    end_date = forms.forms.DateField(label=_('End'), required=False)
 
-    jp_start_date = forms.forms.DateField(label=string_concat(_('Japanese version'), ' - ', _('Beginning')), required=False)
-    jp_end_date = forms.forms.DateField(label=string_concat(_('Japanese version'), ' - ', _('End')), required=False)
+    jp_start_date = forms.forms.DateField(label=string_concat(t['Japanese'], ' ' , _('version'), ' - ', _('Beginning')), required=False)
+    jp_end_date = forms.forms.DateField(label=string_concat(t['Japanese'], ' ' , _('version'), ' - ', _('End')), required=False)
```
Comment (db0company) [inline sukutomo/forms.py]: Same comment as [here](https://github.com/MagiCircles/SchoolIdolTomodachi/pull/28/files#diff-a657129ec4ba09fe870fd26766fd6bccR31), we should use `_('Japanese version')` here and for all the other versions below!

\+ we already have the translated terms for it from BanG Dream so we can reuse them.

File: sukutomo/forms.py [hunk 2]
```diff
@@ -47,24 +48,42 @@ class Meta:
...
 
     class Meta:
         model = models.Event
         save_owner_on_creation = True
+
+class EventFilterForm(MagiFiltersForm):
+    search_fields = ['title']
+
+    ordering_fields = [
+        ('jp_start_date', string_concat(_('Date'), ' (', t['Japanese'], ' ', _('version'), ' )')),
```
Comment (db0company) [inline sukutomo/forms.py]: Same comment regarding using `_('Japanese version')` (+ for all other versions).

File: sukutomo/forms.py [hunk 3]
```diff
@@ -47,24 +48,42 @@ class Meta:
...
-    cn_end_date = forms.forms.DateField(label=string_concat(_('Chinese version'), ' - ', _('End')), required=False)
+    cn_start_date = forms.forms.DateField(label=string_concat(_('Chinese'), ' ' , _('version'), ' - ', _('Beginning')), required=False)
+    cn_end_date = forms.forms.DateField(label=string_concat(_('Chinese'), ' ' , _('version'), ' - ', _('End')), required=False)
 
     class Meta:
         model = models.Event
         save_owner_on_creation = True
+
+class EventFilterForm(MagiFiltersForm):
+    search_fields = ['title']
```
Comment (db0company) [inline sukutomo/forms.py]:
```python
search_fields = ['title', 'd_titles']
```

This way people can search using the name in their own language as well.

File: sukutomo/forms.py [hunk 4]
```diff
@@ -47,24 +48,42 @@ class Meta:
...
+        ('jp_start_date', string_concat(_('Date'), ' (', t['Japanese'], ' ', _('version'), ' )')),
+        ('ww_start_date', string_concat(_('Date'), ' (', _('Worldwide'), ' ',  _('version'), ' )')),
+        ('tw_start_date', string_concat(_('Date'), ' (', _('Taiwanese'), ' ', _('version'), ' )')),
+        ('kr_start_date', string_concat(_('Date'), ' (', _('Korean'), ' ', _('version'), ' )')),
+        ('cn_start_date', string_concat(_('Date'), ' (', _('Chinese'), ' ', _('version'), ' )')),
+        ('title', _('Title')),
+    ]
+
+    version = forms.forms.ChoiceField(label=_(u'Server availability'), choices=BLANK_CHOICE_DASH + models.Account.VERSION_CHOICES)
+    version_filter = MagiFilter(to_queryset=lambda form, queryset, request, value: queryset.filter(c_versions__contains=value))
```
Comment (db0company) [inline sukutomo/forms.py]:
```python
queryset.filter(c_versions__contains=u'"{}"'.format(value))
```

File: sukutomo/models.py [hunk 5]
```diff
@@ -24,16 +24,16 @@ class Account(BaseAccount):
     # Details
 
     VERSIONS = OrderedDict((
-        ('JP', { 'translation': t['Japanese'], 'icon': 'JP' }),
-        ('WW', { 'translation': _('Worldwide'), 'icon': 'world' }),
-        ('KR', { 'translation': t['Korean'], 'icon': 'KR' }),
-        ('CN', { 'translation': t['Chinese'], 'icon': 'CN' }),
-        ('TW', { 'translation': t['Taiwanese'], 'icon': 'TW' }),
+        ('JP', { 'translation': t['Japanese'], 'image': 'ja'}),
```
Comment (db0company) [inline sukutomo/models.py]: Same comment as [there](https://github.com/MagiCircles/SchoolIdolTomodachi/pull/28/files#diff-a657129ec4ba09fe870fd26766fd6bccR31), we should use `_('Japanese version')` here.

File: sukutomo/models.py [hunk 6]
```diff
@@ -273,7 +273,7 @@ class Event(MagiModel):
     FIELDS_PER_VERSION = ['banner', 'countdown', 'start_date', 'end_date']
 
     VERSIONS_CHOICES = Account.VERSION_CHOICES
-    c_versions = models.TextField(_('Server availability'), blank=True, null=True, default='"JP"')
+    c_versions = models.TextField(_('Server availability'), blank=True, null=True, default='JP')
```
Comment (db0company) [inline sukutomo/models.py]: you should keep `"JP"` here because that's the CSV format.

File: sukutomo/forms.py (hunk 1 above)
Comment (LatidoReMe) [reply sukutomo/forms.py]: ✅

File: sukutomo/forms.py (hunk 2 above)
Comment (LatidoReMe) [reply sukutomo/forms.py]: ✅

File: sukutomo/models.py (hunk 5 above)
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/models.py (hunk 6 above)
Comment (LatidoReMe) [reply sukutomo/models.py]: I don't know what CSV format is, but it's done ✅

File: sukutomo/forms.py (hunk 4 above)
Comment (LatidoReMe) [reply sukutomo/forms.py]: ✅

File: sukutomo/forms.py (hunk 3 above)
Comment (LatidoReMe) [reply sukutomo/forms.py]: ✅

File: sukutomo/forms.py [hunk 7]
```diff
@@ -49,40 +49,40 @@ class Meta:
...
+        ('tw_start_date', string_concat(_('Date'), ' (', _('Taiwanese version'), ')')),
+        ('kr_start_date', string_concat(_('Date'), ' (', _('Korean version'), ')')),
+        ('cn_start_date', string_concat(_('Date'), ' (', _('Chinese version'), ')')),
         ('title', _('Title')),
     ]
 
     version = forms.forms.ChoiceField(label=_(u'Server availability'), choices=BLANK_CHOICE_DASH + models.Account.VERSION_CHOICES)
-    version_filter = MagiFilter(to_queryset=lambda form, queryset, request, value: queryset.filter(c_versions__contains=value))
-    
+    version_filter = MagiFilter(to_queryset=lambda form, queryset, request, value: queryset.filter(c_versions__contains=u'{}'.format(value)))
```
Comment (db0company) [inline sukutomo/forms.py]: You forgot the quotes:

```python
u'"{}"'.format(value)
```

The goal is to generate a string between quotes like this: `"en"`

**Pull request merged**